local.tf

```diff
@@ -0,0 +1,566 @@
+}
+
+data "aws_iam_policy_document" "file_metadata_queue_policy" {
+  statement {
+    actions   = ["SQS:SendMessage"]
+    effect    = "Allow"
+    resources = [aws_sqs_queue.file_metadata.arn]
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_sns_topic.file_metadata.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic" "file_metadata" {
+  name = "wandb-file-metadata-topic"
+}
+
+resource "aws_sns_topic_policy" "file_metadata_topic_policy" {
+  arn = aws_sns_topic.file_metadata.arn
+
+  policy = data.aws_iam_policy_document.file_metadata_topic_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_topic_policy" {
+  statement {
+    sid       = "s3-can-publish"
+    actions   = ["SNS:Publish"]
+    effect    = "Allow"
+    resources = [aws_sns_topic.file_metadata.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_s3_bucket.file_storage.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic_subscription" "file_metadata" {
+  topic_arn = aws_sns_topic.file_metadata.arn
+  protocol  = "sqs"
+  endpoint  = aws_sqs_queue.file_metadata.arn
+}
+
+##########################################
+# S3
+##########################################
+
+resource "aws_s3_bucket" "file_storage" {
+  bucket        = "${var.global_environment_name}-wandb-files"
+  acl           = "private"
+  force_destroy = true
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "HEAD"]
+    allowed_origins = ["*"]
+    expose_headers  = ["ETag"]
+    max_age_seconds = 3000
+  }
+
+  server_side_encryption_configuration {
+    rule {
+        apply_server_side_encryption_by_default {
+          sse_algorithm = "AES256"
+        }
+    }
+  }
+}
+
+resource "aws_s3_bucket_notification" "file_metadata_sns" {
+  bucket = aws_s3_bucket.file_storage.id
+
+  topic {
+    topic_arn = aws_sns_topic.file_metadata.arn
+    events    = ["s3:ObjectCreated:*"]
+  }
+}
+
+##########################################
+# RDS
+##########################################
+
+resource "aws_db_subnet_group" "metadata_subnets" {
+  name       = "wandb-db-subnets"
+  subnet_ids = aws_subnet.wandb.*.id
+}
+
+resource "aws_rds_cluster" "metadata_cluster" {
+  engine               = "aurora-mysql"
+  db_subnet_group_name = aws_db_subnet_group.metadata_subnets.name
+
+  skip_final_snapshot     = true
+  backup_retention_period = 14
+
+  enabled_cloudwatch_logs_exports = [
+    "error",
+  ]
+  iam_database_authentication_enabled = true
+
+  database_name   = "wandb_local"
+  master_username = "wandb"
+  master_password = "wandb_root_password"
+
+  vpc_security_group_ids = [aws_security_group.metadata_store.id]
+
+  storage_encrypted = true
+}
```

Comment:
Dbio complained that aurora forced them to have a multi-region HA cluster which added cost.  Would be great to eventually support a non-HA flag that let users have a single DB, but this makes sense as the official template.

Reply:
We should also verify this is the Aurora DB with that supports r-series instances.  FAIR setup an on-demand aurora and we found out that only r-series instances support the parallel query hotness.

Reply:
Forced them? This stack creates an aurora "cluster" (which i believe is sort of just a namespace for many instances) and one writeable instance inside of it. So the billing would just be for the one instance, and it's definitely not highly available or multi-region.

Reply:
There's Amazon Aurora Serverless and then there are dedicated instances.  Apparently parallel queries only work with dedicated r series instances so I want to make sure our default is using the smallest r series or whatever instance size is appropriate instead of the Serverless flavor.

Reply:
The default is r5.large (2 VCPUs, 16G mem)